src/services/UserValidationService.ts

```diff
@@ -1,42 +1,57 @@
 import { Service } from '@tsed/di';
 import { Unauthorized } from '@tsed/exceptions';
-import * as bcrypt from 'bcrypt';
 
-import { User } from '../entities/User';
+import { UserCreation } from '../models/UserCreation';
+import { UserLogin } from '../models/UserLogin';
 
 @Service()
 export class UserValidationService {
 
-	public async validateUser(
-		userToValidate: User, 
-	): Promise<User> {
+	public validateUserLogin(userLogin: UserLogin): void {
+		const email: string= userLogin.email;
+		const password: string = userLogin.password;
+		
+		this.checkParameters([email, password]);
+		this.checkEmail(email);
+		this.checkPassword(password);
+	}
+
+	public validateUserCreation(userCreation: UserCreation): void {
 
-		const username: string = userToValidate.username;
-		const email: string= userToValidate.email;
-		const password: string = userToValidate.password;
+		const username: string = userCreation.username;
+		const email: string= userCreation.email;
+		const password: string = userCreation.password;
 		
-		if (Object.keys(userToValidate).length === 0) {
-			throw new Unauthorized("Empty request body!");
-		}
-	
-		if (!username || !email || !password) {
-			throw new Unauthorized("Missing credentials!");
-		}
+		this.checkParameters([username, email, password]);
+		this.checkUserName(username);
+		this.checkEmail(email);
+		this.checkPassword(password);
+	}
 
+	private checkParameters(parameters: string[]): void {
+		parameters.forEach(s => {
```

Comment:
A different way of writing this and perhaps more clear would be:
```
const hasEmpty = parameters.some(x => !x);
if (hasEmpty) throw new Unathorized("Missing credentials");
```

Reply:
Ok. I will look in to it.